cms/tests/test_app_registration.py

```diff
@@ -53,4 +53,186 @@ def test_imports_cms_apps_files(self):
+        app_list = [app for app in apps.get_app_configs()]
+        self.assertTrue(hasattr(app_list[0], 'cms_app'))
+        self.assertEqual(
+            app_list[0].cms_app.__class__.__name__, 'CMSSomeFeatureConfig')
+        self.assertFalse(hasattr(app_list[1], 'cms_app'))
+
+    @override_settings(INSTALLED_APPS=[
+        'cms.tests.test_app_registry.app_without_cms_app_class',
+    ])
+    def test_raises_exception_when_no_cms_app_class_found_in_cms_file(self):
+        with self.assertRaises(ImproperlyConfigured):
+            app_registration.autodiscover_cms_configs()
+
+    @override_settings(INSTALLED_APPS=[
+        'cms.tests.test_app_registry.app_with_two_cms_app_classes',
+    ])
+    def test_raises_exception_when_more_than_one_cms_app_class_found_in_cms_file(self):
+        with self.assertRaises(ImproperlyConfigured):
+            app_registration.autodiscover_cms_configs()
+
+
+class GetCmsAppsWithFeaturesTestCase(CMSTestCase):
+
+    @patch.object(apps, 'get_app_configs')
+    def test_returns_only_apps_with_features(self, mocked_apps):
+        # apps with cms_app attr and a configure method
+        app_with_cms_ext1 = Mock(cms_app=Mock(spec=['configure_app']))
+        app_with_cms_ext2 = Mock(cms_app=Mock(spec=['configure_app']))
+        # app with cms_app attr without a configure method
+        # this throws an AttributeError if configure_app is accessed
+        app_with_cms_config = Mock(cms_app=Mock(spec=[]))
+        # app without cms_app attr
+        # this throws an AttributeError if cms_app is accessed
+        non_cms_app = Mock(spec=[])
+        # mock what apps have been installed
+        mocked_apps.return_value = [
+            app_with_cms_ext1,
+            app_with_cms_config,
+            app_with_cms_ext2,
+            non_cms_app,
+        ]
+
+        apps_with_features = app_registration.get_cms_apps_with_features()
+
+        self.assertListEqual(
+            apps_with_features,
+            [app_with_cms_ext1, app_with_cms_ext2])
+
+
+class ConfigureCmsAppsTestCase(CMSTestCase):
+
+    @patch.object(apps, 'get_app_configs')
+    def test_runs_configure_app_method_for_app_with_enabled_config(
+        self, mocked_apps
+    ):
+        # Set up app with name djangocms_feature_x that has a cms feature
+        feature_app = Mock(spec=AppConfig)
+        feature_app.name = 'djangocms_feature_x'
+        feature_app.cms_app = Mock(spec=['configure_app'])
+        # Set up app that makes use of djangocms_feature_x
+        config_app_enabled = Mock(spec=AppConfig)
+        config_app_enabled.cms_app = Mock(
+            spec=['djangocms_feature_x_enabled'])
+        config_app_enabled.cms_app.djangocms_feature_x_enabled = True
+        # Pretend these mocked apps are in INSTALLED_APPS
+        mocked_apps.return_value = [
+            feature_app, config_app_enabled]
+
+        app_registration.configure_cms_apps([feature_app])
+
+        feature_app.cms_app.configure_app.assert_called_once_with(
```

Comment:
What is this assertion doing? A small comment would make it easy for me to understand. Again personal opinion.